### PR TITLE
Add calculation for faerie fire feral

### PR DESCRIPTION
**Pull request by waiwai2233**
This is my first time contributing to such a large project and I understand that I may have made some mistakes.

This pull request contains the following changes related to fearie fire (bear form):

Added "Curse of Elements" effect in damage debuffs
Added "Misery" effect in stat debuffs
Added spell hit calculation in main.py, taking into account debuffs like misery. Player's hit is calculated using the formula `buffed_spell_hit=unbuffed_spell_hit + 1 * ('heroic_presence' in raid_buffs) + 40 / 26.23 * ('hit_food' in consumables)` Encounter hit is calculated using the formula `encounter_spell_hit = (buffed_spell_hit + 3 * ('misery' in stat_debuffs))`
Added spell multiplier, taking into account Heart of the Crusader / Master Poisoner
Modified the faerie_fire() function in the Player class to account for the damage done by Faerie Fire in bear form using formula `faerie_dmg = (0.15 * bear_ap + 1.) * (1 + 0.13 * curse_of_elements) * spell_damage_multiplier`

Please review my changes and let me know if there are any issues or concerns and I look forward to contributing more to this project in the future.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -271,8 +273,9 @@
                  'value': 'jotc'
              },
              {'label': 'Judgment of Wisdom', 'value': 'jow'},
+             {'label': 'Misery / Improved Fearie Fire', 'value': 'misery'},
```
Comment (NerdEgghead) [inline main.py]: Small typo, should be "Faerie" not "Fearie"

File: player.py [hunk 2]
```diff
@@ -51,6 +51,8 @@ def __init__(
             ap_mod (float): Total multiplier for Attack Power in Cat Form.
             agility (int): Fully raid buffed Agility attribute.
             hit_chance (float): Chance to hit as a fraction.
+            spell_hit_chance (float): Chance for spells (farie fire feral)
```
Comment (NerdEgghead) [inline player.py]: small typo again, should be "Faerie Fire" not "farie fire"

File: player.py [hunk 3]
```diff
@@ -1022,9 +1033,18 @@ def faerie_fire(self):
...
-        return 0.0
+        if self.cat_form:
+            self.dmg_breakdown['Faerie Fire (Cat)']['casts'] += 1
+            if self.log:
+                self.gen_log('Faerie Fire (Cat)', '', False, False, False)
+            return 0.0
+
+        self.dmg_breakdown['Faerie Fire (Bear)']['casts'] += 1
+        miss = (np.random.rand() < self.spell_miss_chance)
+        damage_on_hit = self.faerie_fire_hit
```
Comment (NerdEgghead) [inline player.py]: Faerie Fire (Bear) can crit, so this calculation is insufficient. You'll need to store the player's spell crit as well and then roll for crits using the calc_yellow_damage() helper function.

File: player.py [hunk 4]
```diff
@@ -1022,9 +1033,18 @@ def faerie_fire(self):
...
+        self.gen_log('Faerie Fire (Bear)', damage_on_hit, miss, False, False)
```
Comment (NerdEgghead) [inline player.py]: Should wrap this in an `if self.log:` block for consistency with the rest of the code.

File: player.py [hunk 5]
```diff
@@ -202,6 +206,8 @@ def calc_miss_chance(self):
             (8. - miss_reduction) + (6.5 - dodge_reduction)
         )
         self.dodge_chance = 0.01 * (6.5 - dodge_reduction)
+        spell_miss_reduction = min(self.spell_hit_chance * 100, 17.0)
+        self.spell_miss_chance = 0.01 * (17.0 - spell_miss_reduction)
```
Comment (NerdEgghead) [inline player.py:214]: What's missing here is that dynamic changes to Hit Rating during a fight (from trinket procs etc.) or more importantly, Hit Rating adjustments within `calc_stat_weights`, won't correctly propagate to the spell miss chance.

**Review by NerdEgghead — CHANGES_REQUESTED** [review]
Thanks for the contribution! Looks good as a starting point, but the two important missing pieces are (a) propagating dynamic Hit Rating changes to the stored spell miss chance, and (b) implementing spell crits on Faerie Fire.

File: main.py [hunk 6]
```diff
@@ -258,8 +258,10 @@
                  'label': 'Blood Frenzy / Savage Combat',
                  'value': 'blood_frenzy'
              },
+             {'label': "Curse of Elements / Earth and Moon / Ebon Plaguebringer",
+              'value': 'curse_of_elements'},
```
Comment (Santiclause) [inline main.py]: for a multiline dict like this, you should format it like the dict above
```suggestion
             {
                 'label': "Curse of Elements / Earth and Moon / Ebon Plaguebringer",
                 'value': 'curse_of_elements'
             },
```

File: main.py [hunk 7]
```diff
@@ -271,8 +273,12 @@
                  'value': 'jotc'
              },
              {'label': 'Judgment of Wisdom', 'value': 'jow'},
+             {'label': 'Misery / Improved Faerie Fire', 'value': 'misery'},
+             {'label': 'Shadow Mastery / Improved Scorch / Winter\'s Chill',
+              'value': 'shadow_mastery'
```
Comment (Santiclause) [inline main.py]: same suggestion
```suggestion
             {
                 'label': 'Shadow Mastery / Improved Scorch / Winter\'s Chill',
                 'value': 'shadow_mastery'
```

**Review by Santiclause — COMMENTED** [review]
some pitnicky suggestions - also, I don't see 5% spell crit aura from boomie/ele sham listed anywhere. not sure if that oversight is intentional or not

File: main.py (hunk 7 above)
Comment (waiwai2233) [reply main.py]: okok I will change it right now, btw, do you know where i can find info for spell resistance?

**Pull request merged**